src/dataset/impl_dataset.rs

```diff
@@ -1091,6 +1091,32 @@ where
     }
 }
 
+impl<F: Float, D, S, O> PredictRef<ArrayBase<D, Ix2>, Array1<S>> for O
```

Comment:
Once we finalize the new trait we can cut out `PredictRef` and define `Predict` in terms of `PredictInto` instead.

Reply:
Good idea. I will remove `PredictRef` after the other changes are done.